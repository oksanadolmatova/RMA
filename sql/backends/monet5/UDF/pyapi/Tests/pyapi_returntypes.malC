
r0:bat[:int] := pyapi.eval(nil:ptr,"return(33)");
io.print(r0);

# return a single float value
r6:bat[:flt] := pyapi.eval(nil:ptr,"return(12.31)");
io.print(r6);

# return a single string value
r7:bat[:str] := pyapi.eval(nil:ptr,"return(\"test\")");
io.print(r7);

#####################
# --PYTHON ARRAYS-- #
#####################
# return an array with a single value
r1:bat[:int] := pyapi.eval(nil:ptr,"return([33])");
io.print(r1);

# return an array containing a single array
r2:bat[:int] := pyapi.eval(nil:ptr,"return([[33]])");
io.print(r2);

# return an array containing two arrays
(r3:bat[:int], s3:bat[:int]) := pyapi.eval(nil:ptr,"return([[33], [42]])");
io.print(r3, s3);

# return an array containing three arrays
(r4:bat[:int], s4:bat[:int], t4:bat[:int]) := pyapi.eval(nil:ptr,"return([[33], [42], [66]])");
io.print(r4, s4, t4);

# return an array containing two arrays containing strings
(r5:bat[:str], s5:bat[:str]) := pyapi.eval(nil:ptr,"return([[\"Hello\", \"Hello Again\"], [\"Hello Again Again\",\"That's quite enough.\"]])");
io.print(r5,s5);


####################
# --NUMPY ARRAYS-- #
####################
# return a single numpy array
r8:bat[:int] := pyapi.eval(nil:ptr,"return(numpy.array([33,11,332,51,88]))");
io.print(r8);

# return an array containing two numpy arrays
(r9:bat[:int], s9:bat[:int]) := pyapi.eval(nil:ptr,"return([numpy.array([33]), numpy.array([44])])");
io.print(r9, s9);

# return a numpy array containing two arrays of unequal size
(r10:bat[:int], s10:bat[:int]) := pyapi.eval(nil:ptr,"return(numpy.array([[33,24,55], [44,66, 345, 77, 66]]))");
io.print(s10);

# return a numpy array containing two numpy arrays of unequal size
(r11:bat[:int], s11:bat[:int]) := pyapi.eval(nil:ptr,"return(numpy.array([numpy.array([33]), numpy.array([44, 45, 77])]))");
io.print(s11);

# return a multi-dimensional numpy array containing integers
(r12:bat[:int], s12:bat[:int]) := pyapi.eval(nil:ptr,"return(numpy.array([[33, 24, 55], [44, 66,345]]))");
io.print(r12,s12);

# return a multi-dimensional numpy array containing doubles 
(r13:bat[:flt], s13:bat[:flt]) := pyapi.eval(nil:ptr,"return(numpy.array([[27.42, 18.62, 33.81], [12.55, 41.44, 20.88]]))");
io.print(r13,s13);

# return a multi-dimensional numpy array containing unicode strings
bstr:= bat.new(:str);
bat.append(bstr,"Hannes Mühleisen":str);
bat.append(bstr,"héllo":str);
bat.append(bstr,"éáú üüäãö":str);
bat.append(bstr,"幺巾乡阜阝测试一些中国符号":str);
bat.append(bstr,"いくつかの日本のシンボルをテストします":str);
(z123:bat[:str], z234:bat[:str]) := pyapi.eval(nil:ptr,"return(numpy.array([arg1, arg1]))",bstr);
io.print(z123,z234);

# return a multi-dimensional numpy array containing integers, but store them in a string BAT
(r14:bat[:str], s14:bat[:str]) := pyapi.eval(nil:ptr,"return(numpy.array([[33, 24, 55], [44, 66,345]]))");
io.print(r14,s14);

# return a multi-dimensional python array and load it into different types of BATs
(di:bat[:int],dd:bat[:flt]) := pyapi.eval(nil:ptr,"return(numpy.array([[33.2, 24.7, 55.4], [44.1, 66.3, 2.2]]))");
io.print(di,dd);

#####################
# --MASKED ARRAYS-- #
#####################
# return a single masked array
r15:bat[:int] := pyapi.eval(nil:ptr, "return(numpy.ma.masked_array([862,43,24],[1,1,0]))");
io.print(r15);

# return a single masked array in an array
r16:bat[:int] := pyapi.eval(nil:ptr, "return([numpy.ma.masked_array([862,43,24],[1,1,0])])");
io.print(r16);

# return a single masked array in a numpy array (this is kind of weird, it removes the entire masked array so it's a single numpy array)
r17:bat[:int] := pyapi.eval(nil:ptr, "return(numpy.array([numpy.ma.masked_array([862,43,24],[1,1,0])]))");
io.print(r17);

# return a single masked array containing strings
r18:bat[:str] := pyapi.eval(nil:ptr, "return(numpy.ma.masked_array([\"Test\",\"Four\",24],[0,0,1]))");
io.print(r18);

# return a multi-dimensional masked array with even dimensions
(r19:bat[:int], s19:bat[:int]) := pyapi.eval(nil:ptr,"return(numpy.ma.masked_array([[1, 2, 3], [1,2,3]], [[0,0,0], [1,0,0]]))");
io.print(r19,s19);

# return a multi-dimensional masked array with even dimensions
(r20:bat[:str], s20:bat[:str]) := pyapi.eval(nil:ptr,"return(numpy.ma.masked_array([['str', 'test', 'k'], ['test','test213','test44']], [[0,0,0], [1,0,0]]))");
io.print(r20,s20);
