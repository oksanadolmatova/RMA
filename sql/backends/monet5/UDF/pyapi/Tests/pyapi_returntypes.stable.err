stderr of test 'pyapi_returntypes` in directory 'sql/backends/monet5/UDF/pyapi` itself:


# 23:33:07 >  
# 23:33:07 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=36739" "--set" "mapi_usock=/var/tmp/mtest-20340/.s.monetdb.36739" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/home/mytherin/opt/var/mTests_monetdb5_extras_pyapi" "--set" "mal_listing=2" "--set" "embedded_r=true" "--set" "embedded_py=true"
# 23:33:07 >  

# builtin opt 	gdk_dbpath = /home/mytherin/opt/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 36739
# cmdline opt 	mapi_usock = /var/tmp/mtest-20340/.s.monetdb.36739
# cmdline opt 	monet_prompt = 
# cmdline opt 	mal_listing = 2
# cmdline opt 	gdk_dbpath = /home/mytherin/opt/var/mTests_monetdb5_extras_pyapi
# cmdline opt 	mal_listing = 2
# cmdline opt 	embedded_r = true
# cmdline opt 	embedded_py = true
# cmdline opt 	gdk_debug = 536870922

# 23:33:07 >  
# 23:33:07 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-20340" "--port=36739"
# 23:33:07 >  


# 23:33:07 >  
# 23:33:07 >  "Done."
# 23:33:07 >  

