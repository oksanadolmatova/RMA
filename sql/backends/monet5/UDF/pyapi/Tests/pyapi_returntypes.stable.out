stdout of test 'pyapi_returntypes` in directory 'sql/backends/monet5/UDF/pyapi` itself:


# 23:33:07 >  
# 23:33:07 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=36739" "--set" "mapi_usock=/var/tmp/mtest-20340/.s.monetdb.36739" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/home/mytherin/opt/var/mTests_monetdb5_extras_pyapi" "--set" "mal_listing=2" "--set" "embedded_r=true" "--set" "embedded_py=true"
# 23:33:07 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_extras_pyapi', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 7.684 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://mytherin-N750JV:36739/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-20340/.s.monetdb.36739
# MonetDB/SQL module loaded
# MonetDB/Python module loaded
# MonetDB/R   module loaded

Ready.

# 23:33:07 >  
# 23:33:07 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-20340" "--port=36739"
# 23:33:07 >  

#io.print(r);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	33	]
#io.print(r);
#--------------------------#
# h	t  # name
# void	flt  # type
#--------------------------#
[ 0@0,	12.31	]
#io.print(r);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"test"	]
#io.print(r);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	33	]
#io.print(r);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	33	]
#io.print(r, s);
#--------------------------#
# h	t	t  # name
# void	int	int  # type
#--------------------------#
[ 0@0,	33,	42	]
#io.print(r, s, t);
#--------------------------#
# h	t	t	t  # name
# void	int	int	int  # type
#--------------------------#
[ 0@0,	33,	42,	66	]
#io.print(r,s);
#--------------------------#
# h	t	t  # name
# void	str	str  # type
#--------------------------#
[ 0@0,	"Hello",	"Hello Again Again"	]
[ 1@0,	"Hello Again",	"That's quite enough."	]
#io.print(r);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	33	]
[ 1@0,	11	]
[ 2@0,	332	]
[ 3@0,	51	]
[ 4@0,	88	]
#io.print(r, s);
#--------------------------#
# h	t	t  # name
# void	int	int  # type
#--------------------------#
[ 0@0,	33,	44	]
#io.print(s);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	44	]
[ 1@0,	66	]
[ 2@0,	345	]
[ 3@0,	77	]
[ 4@0,	66	]
#io.print(s);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	44	]
[ 1@0,	45	]
[ 2@0,	77	]
#io.print(r,s);
#--------------------------#
# h	t	t  # name
# void	int	int  # type
#--------------------------#
[ 0@0,	33,	44	]
[ 1@0,	24,	66	]
[ 2@0,	55,	345	]
#io.print(r,s);
#--------------------------#
# h	t	t  # name
# void	flt	flt  # type
#--------------------------#
[ 0@0,	27.42,	12.55	]
[ 1@0,	18.62,	41.44	]
[ 2@0,	33.81,	20.88	]
#io.print(r,s);
#--------------------------#
# h	t	t  # name
# void	str	str  # type
#--------------------------#
[ 0@0,	"Hannes Mühleisen",	"Hannes Mühleisen"	]
[ 1@0,	"héllo",	"héllo"	]
[ 2@0,	"éáú üüäãö",	"éáú üüäãö"	]
[ 3@0,	"幺巾乡阜阝测试一些中国符号",	"幺巾乡阜阝测试一些中国符号"	]
[ 4@0,	"いくつかの日本のシンボルをテストします",	"いくつかの日本のシンボルをテストします"	]
#io.print(r,s);
#--------------------------#
# h	t	t  # name
# void	str	str  # type
#--------------------------#
[ 0@0,	"33",	"44"	]
[ 1@0,	"24",	"66"	]
[ 2@0,	"55",	"345"	]
#--------------------------#
# h	t	t  # name
# void	int	flt  # type
#--------------------------#
[ 0@0,	33,	44.1	]
[ 1@0,	24,	66.3	]
[ 2@0,	55,	2.2	]
#io.print(r);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	nil	]
[ 1@0,	nil	]
[ 2@0,	24	]
#io.print(r);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	nil	]
[ 1@0,	nil	]
[ 2@0,	24	]
#io.print(r);
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	862	]
[ 1@0,	43	]
[ 2@0,	24	]
#io.print(r);
#--------------------------#
# h	t  # name
# void	str  # type
#--------------------------#
[ 0@0,	"Test"	]
[ 1@0,	"Four"	]
[ 2@0,	nil	]
#io.print(r,s);
#--------------------------#
# h	t	t  # name
# void	int	int  # type
#--------------------------#
[ 0@0,	1,	nil	]
[ 1@0,	2,	2	]
[ 2@0,	3,	3	]
#io.print(r,s);
#--------------------------#
# h	t	t  # name
# void	str	str  # type
#--------------------------#
[ 0@0,	"str",	nil	]
[ 1@0,	"test",	"test213"	]
[ 2@0,	"k",	"test44"	]

# 23:33:07 >  
# 23:33:07 >  "Done."
# 23:33:07 >  

